cpp/arcticdb/storage/s3/s3_mock_client.hpp

```diff
@@ -67,9 +71,9 @@ class MockS3Client : public S3ClientWrapper {
             const std::string& prefix,
             const std::string& bucket_name,
             const std::optional<std::string> continuation_token) const override;
+
 private:
-    // Stores a mapping from pair<bucket_name, s3_name> to a Segment.
-    std::map<std::pair<std::string, std::string>, Segment> s3_contents;
+    std::map<S3Key, Segment> s3_contents;
```

Comment:
Don't understand why this is an ordered map?

Reply:
1. The minor advantage is that when you run a `list_objects` on it, it will return a sorted list where all the keys are already lexicographically sorted. In [places like this](https://github.com/man-group/ArcticDB/blob/master/cpp/arcticdb/storage/test/test_s3_storage.cpp#L281C1-L285C6) it can be handy if you decide to print the list. (You could sort the returned list manually though if you use `std::unordered_map`
2. [`std::unordered_map`](https://en.cppreference.com/w/cpp/container/unordered_map) would require defining a hash function, the `operator<` and `operator==` for `S3Key` so `std::map` is slightly easier to setup. My reasoning for sticking to `std::map` is that we are not aiming for testing mock client with a huge number of symbols where `std::unordered_map` would be advantageous from a performance point of view. The maximum number of symbols we have tested a mock client with is 25.